src/playMiddleware.js

```diff
@@ -0,0 +1,50 @@
+import videojs from 'video.js';
+
+const obj = {};
+
+/**
+ * Checks if middleware mediators are available and
+ * can be used on this platform.
+ * Currently we can only use mediators on desktop platforms.
+ */
+obj.isMiddlewareMediatorSupported = function() {
+
+  if (videojs.browser.IS_IOS || videojs.browser.IS_ANDROID) {
+    return false;
+
+  } else if (
+    // added when middleware was introduced in video.js
+    videojs.use &&
+    // added when mediators were introduced in video.js
+    videojs.middleware &&
+    videojs.middleware.TERMINATOR) {
+    return true;
+
+  }
+
+  return false;
+};
+
+obj.playMiddleware = function(player) {
+  return {
+    setSource(srcObj, next) {
+      next(null, srcObj);
+    },
+    callPlay() {
+      // Block play calls while waiting for an ad
+      if (obj.isMiddlewareMediatorSupported() &&
```

Comment:
Is the check to `isMiddlewareMediatorSupported` needed?

Reply:
I ask because it's checked before setting up the middleware in plugin.js

Reply:
This is mostly defensive. Although it is checked before registering the the middleware, if in case that changes we want to make sure we don't use both the playMiddleware and cancelContentPlay at the same time.

Reply:
I think in this case it doesn't have much defensive value but could lead to confusion about how the code works. That's subjective though so not a big deal.

Reply:
So here's my thinking behind the "defensive code" reasoning:
- once a middleware is registered, it will always run(since we're using the MIME type agnostic '*')
- there's a number of video.js versions that don't have Middleware Mediator support (since it was added relatively recently)
- since the middleware is registered in plugin.js, I'm worried that the reason why we check  `isMiddlewareMediatorSupported` would be lost over time and we'll wind up doing something that causes a runtime exception.

Maybe I'm just worrying about nothing though.

Reply:
Once middleware is registered it will always run, and conditions for it being registered will never change, so it won't need to be checked again.

The video.js versions that don't have mediator support are handled within the check before registering.

The last concern is definitely a valid one, but I think it is best addressed by commenting the code to add the reason. Basically it sounds like you're writing a workaround for a hypothetical bug, but really, you're making it harder to identify the hypothetical bug when if and when it happens.

Reply:
Maybe the last concern can be addressed by doing two things:
- adding more comments to explain why we check for support first in plugin.js
- check if videojs.middleware exists before trying to return the middleware.TERMINATOR here, to avoid a runtime exception breaking the whole player. 

Also, this check is also done in `initCancelContentPlay`, do you feel the same about that?

Reply:
"adding more comments to explain why we check for support first in plugin.js" 👍 

"check if videojs.middleware exists before trying to return the middleware.TERMINATOR" That doesn't seem much different. If middleware isn't supported, we would not have gotten that far.

"Also, this check is also done in initCancelContentPlay, do you feel the same about that?" I do, yeah. It's performing a check for an unrelated feature. The logic should be hoisted to the first place that can handle it and form a bottleneck rather than spreading checks throughout various modules.